### ISSUE TITLE
Does the training of retriever and reader share the same dev set?

### ISSUE THREAD

**Issue by wuyaoxuehun** [facebookresearch/DPR#140]
Does the training of retriever and reader share the same train and dev set? 

If so, will this lead to overfitting, since the reader uses the same train and dev set as the retriever, so the input passages  from the DPR retriever for the reader will be too ideal during training and validating while the input passages for the test set will not be so ideal?

Thanks!

**Comment by vlad-karpukhin** [comment]
reader and retriever share almost the same train/dev set questions, positive passages, but not hard negatives passages.

I don't see why it should lead to overfitting since the models are completely independent.

In fact, using ideal 'gold' passages for reader instead of distantly supervised positives returned from the retriever  improves test results slightly.

**Comment by wuyaoxuehun** [comment]
> 
> 
> reader and retriever share almost the same train/dev set questions, positive passages, but not hard negatives passages.
> 
> I don't see why it should lead to overfitting since the models are completely independent.
> 
> In fact, using ideal 'gold' passages for reader instead of distantly supervised positives returned from the retriever improves test results slightly.

Thanks for your reply! It helps a lot!
